### PR TITLE
Replace nbsp in all rustdoc code blocks

### LINKED ISSUE #106098: Use of &nbsp; by rustdoc makes APIs not copyable into Rust source code
Suppose we have this trait:

```rust
// lib.rs

pub trait Trait {
    fn method<T>()
    where
        T: Default;
}
```

Run `cargo doc`; it renders as:

Now let's begin writing an impl of this trait:

```rust
struct MyStruct;

impl Trait for MyStruct {
}
```

and copy in the `method` method signature directly from the documentation:

```rust
struct MyStruct;

impl Trait for MyStruct {
    fn method<T>()
    where
        T: Default {}
}
```

We get this distressing error from `cargo check`:

```console
error: unknown start of token: \u{a0}
  --> src/main.rs:11:1
   |
11 |     where
   | ^
   |
help: Unicode character ' ' (No-Break Space) looks like ' ' (Space), but it is not
   |
11 |     where
   | +

error: unknown start of token: \u{a0}
  --> src/main.rs:11:2
   |
11 |     where
   |  ^
   |
help: Unicode character ' ' (No-Break Space) looks like ' ' (Space), but it is not
   |
11 |     where
   |  +

error: unknown start of token: \u{a0}
  --> src/main.rs:11:3
   |
11 |     where
   |   ^
   |
help: Unicode character ' ' (No-Break Space) looks like ' ' (Space), but it is not
   |
11 |     where
   |   +

error: unknown start of token: \u{a0}
  --> src/main.rs:11:4
   |
11 |     where
   |    ^
   |
help: Unicode character ' ' (No-Break Space) looks like ' ' (Space), but it is not
   |
11 |     where
   |    +

error: unknown start of token: \u{a0}
  --> src/main.rs:12:1
   |
12 |         T: Default {}
   | ^
   |
help: Unicode character ' ' (No-Break Space) looks like ' ' (Space), but it is not
   |
12 |         T: Default {}
   | +

error: unknown start of token: \u{a0}
  --> src/main.rs:12:2
   |
12 |         T: Default {}
   |  ^
   |
help: Unicode character ' ' (No-Break Space) looks like ' ' (Space), but it is not
   |
12 |         T: Default {}
   |  +

error: unknown start of token: \u{a0}
  --> src/main.rs:12:3
   |
12 |         T: Default {}
   |   ^
   |
help: Unicode character ' ' (No-Break Space) looks like ' ' (Space), but it is not
   |
12 |         T: Default {}
   |   +

error: unknown start of token: \u{a0}
  --> src/main.rs:12:4
   |
12 |         T: Default {}
   |    ^
   |
help: Unicode character ' ' (No-Break Space) looks like ' ' (Space), but it is not
   |
12 |         T: Default {}
   |    +

error: unknown start of token: \u{a0}
  --> src/main.rs:12:5
   |
12 |         T: Default {}
   |     ^
   |
help: Unicode character ' ' (No-Break Space) looks like ' ' (Space), but it is not
   |
12 |         T: Default {}
   |     +

error: unknown start of token: \u{a0}
  --> src/main.rs:12:6
   |
12 |         T: Default {}
   |      ^
   |
help: Unicode character ' ' (No-Break Space) looks like ' ' (Space), but it is not
   |
12 |         T: Default {}
   |      +

error: unknown start of token: \u{a0}
  --> src/main.rs:12:7
   |
12 |         T: Default {}
   |       ^
   |
help: Unicode character ' ' (No-Break Space) looks like ' ' (Space), but it is not
   |
12 |         T: Default {}
   |       +

error: unknown start of token: \u{a0}
  --> src/main.rs:12:8
   |
12 |         T: Default {}
   |        ^
   |
help: Unicode character ' ' (No-Break Space) looks like ' ' (Space), but it is not
   |
12 |         T: Default {}
   |        +
```

Hopefully there is some other way rustdoc would be able to display the same content in a way that doesn't suffer this limitation.

Is this just as simple as using `white-space: pre;` in the CSS and then writing ordinary spaces instead of &amp;nbsp;? If so, that seems definitely worthwhile doing.

### PR DESCRIPTION
**Pull request by notriddle**
Based on #106125 by @dtolnay — this PR fixes the line wrapping bug.

Fixes #106098. This makes code copyable from rustdoc rendered documentation into a Rust source file.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
Thanks!

@bors r+ rollup

File: src/librustdoc/html/format.rs [hunk 1]
```diff
@@ -368,7 +368,7 @@ pub(crate) fn print_where_clause<'a, 'tcx: 'a>(
                     format!("<br><span class=\"where\">where{where_preds}</span>")
                 } else {
                     let mut clause = br_with_padding;
-                    clause.truncate(clause.len() - 4 * "&nbsp;".len());
+                    clause.truncate(clause.len() - 4 * " ".len());
```
Comment (kadiwa4) [inline src/librustdoc/html/format.rs:371]: maybe remove the `* " ".len()` entirely

Comment (GuillaumeGomez) [reply src/librustdoc/html/format.rs:371]: Outch ood catch!

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks! r=me once CI pass.

**Pull request merged**